src/ol/style/webgl.js

```diff
@@ -21,7 +20,15 @@
 
 /**
  * @typedef {Object} FillProps
- * @property {ColorExpression} [fill-color] The fill color.
+ * @property {ColorExpression} [fill-color] Fill color.
+ * @property {string} [fill-pattern-src] Fill pattern image source URI. If `fill-color` is defined as well, it will be used to tint this image.
+ * @property {HTMLImageElement|HTMLCanvasElement} [fill-pattern-img] Fill pattern image object. Can be provided as an alternative to `fill-pattern-src`.
```

Comment:
Accepting an image here is at odds with the goal of parsing styles and preparing attribute buffers in a worker.  But we could wait until we are doing that and then deal with this then.

Reply:
Absolutely, and a procedurally-generated image could easily be made into an object URL that could then be read from a worker.

I really went with a conservative approach here and mimicked the existing icon properties (also true for the `origin-offset` properties which @ahocevar mentioned). But I'm considering removing the code path for `Image` objects entirely, as there's no point in introducing APIs which we know are not aligned with long-term goals anyway.